src/Doctrine/DBAL/Types/PhoneNumberType.php

```diff
@@ -50,12 +50,17 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      */
     public function convertToDatabaseValue($value, AbstractPlatform $platform)
     {
-        if (null === $value) {
+        if (!(bool) $value) {
             return null;
         }
 
         if (!$value instanceof PhoneNumber) {
-            throw new ConversionException('Expected \libphonenumber\PhoneNumber, got '.\gettype($value));
+            // BC between "doctrine/doctrine-bundle": "^1.12" and "^2.0"
+            if (method_exists(ConversionException::class, 'conversionFailedInvalidType')) {
+                throw ConversionException::conversionFailedInvalidType($value, $this->getName(), ['null', PhoneNumber::class]);
+            } else {
+                throw new ConversionException(sprintf('Expected null or %s, got %s', PhoneNumber::class, \gettype($value)));
+            }
```

Comment:
This change is interesting... But has nothing to do with doctrine bundle. It actually has been introduced by doctrine/dbal:2.6.0

Reply:
It's still not related to any doctrine bundle version but doctrine dbal. Even if it's not directly written in the composer.json file.

To be more clear doctrine bundle v1.x is compatible with doctrine/dbal 2.6+ as well as doctrine bundle v2.x. But doctrine bundle 2.x enforce dbal 2.9+, that's right, but the BC is not related to the bundle itself since it can work with dbal 2.6+ no matter its version.